.buildkite/scripts/validate_clang_format.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```

Comment:
It is probably prudent to maintain posix shell compliance so that we can be as flexible as possible. 

This means `export -f` wont exist so we need to rework this script (and the other one).

Is it worth having two separate scripts/build processes to do the validation? Why not just merge all validation into a single script?